pallets/proofs-dealer/src/benchmarking.rs

```diff
@@ -1,15 +1,53 @@
 //! Benchmarking setup for pallet-proofs-dealer
+
 use frame_benchmarking::v2::*;
 
-#[benchmarks]
+#[benchmarks(
+    where
+        // Runtime `T` implements, `pallet_balances::Config` `pallet_storage_providers::Config` and this pallet's `Config`.
+        T: pallet_balances::Config + pallet_storage_providers::Config + crate::Config,
+        // The Storage Providers pallet is the `Providers` pallet that this pallet requires.
+        T: crate::Config<ProvidersPallet = pallet_storage_providers::Pallet<T>>,
+        // The `Balances` pallet is the `NativeBalance` pallet that this pallet requires.
+        T: crate::Config<NativeBalance = pallet_balances::Pallet<T>>,
+        // The `Balances` pallet is the `NativeBalance` pallet that `pallet_storage_providers::Config` requires.
+        T: pallet_storage_providers::Config<NativeBalance = pallet_balances::Pallet<T>>,
+        // The `Proof` inner type of the `ForestVerifier` trait is `CompactProof`.
+        <T as crate::Config>::ForestVerifier: shp_traits::CommitmentVerifier<Proof = sp_trie::CompactProof>,
+        // The `Proof` inner type of the `KeyVerifier` trait is `CompactProof`.
+        <<T as crate::Config>::KeyVerifier as shp_traits::CommitmentVerifier>::Proof: From<sp_trie::CompactProof>,
+        // The Storage Providers pallet's `HoldReason` type can be converted into the Native Balance's `Reason`.
+        pallet_storage_providers::HoldReason: Into<<<T as pallet::Config>::NativeBalance as frame_support::traits::fungible::InspectHold<<T as frame_system::Config>::AccountId>>::Reason>,
+        // The Storage Providers `MerklePatriciaRoot` type is the same as `frame_system::Hash`.
+        T: pallet_storage_providers::Config<MerklePatriciaRoot = <T as frame_system::Config>::Hash>,
+        // The Storage Providers `ProviderId` type is the same as `frame_system::Hash`.
+        T: pallet_storage_providers::Config<ProviderId = <T as frame_system::Config>::Hash>,
```

Comment:
these comments are a little verbose imo, it's understandable just by reading the generic bounds

Reply:
Well maybe for us 'cause we've been reading these types and pallets for so long 😂